test/server-test.js

```diff
@@ -1,52 +1,189 @@
+        code.expect(e.toString()).to.equal('Error: Authentication strategy name already exists');
+        success = true;
+      }
+      code.expect(success).to.equal(true);
+      done();
+    });
+  });
+});
+
+lab.test('strategy prevents access of protected routes ', (done) => {
+  server.register({
+    register: strategyLoader,
+    options: config
+  }, (err) => {
+    if (err) {
+      console.log(err);
+    }
+    server.route({
+      method: 'GET',
+      path: '/',
+      config: {
+        auth: 'session',
+        handler: (request, reply) => {
+          reply('hello!');
+        }
+      }
+    });
+    server.start(() => {
+      server.inject({
+        url: '/',
+      }, (res) => {
+        code.expect(res.statusCode).to.equal(401);
+        done();
+      });
+    });
+  });
+});
+
+lab.test('hapi-auth-cookie can set a cookie key', (done) => {
+  server.register({
+    register: strategyLoader,
+    options: config
+  }, (err) => {
+    if (err) {
+      console.log(err);
     }
-    server.register({
-      register: strategyLoader,
-      options: config
-    }, (err) => {
-      if (err) {
-        console.log(err);
-      }
-      server.start((startErr, serv) => {
-        let success = false;
-        try {
-          server.auth.strategy('session', 'cookie', 'try', {});
-        } catch (e) {
-          code.expect(e.toString()).to.equal('Error: Authentication strategy name already exists');
-          success = true;
+    server.route({
+      method: 'GET',
+      path: '/',
+      config: {
+        auth: 'session',
+        handler: (request, reply) => {
+          reply('hello!');
         }
-        code.expect(success).to.equal(true);
+      }
+    });
+    server.route({
+      method: 'GET', path: '/login/{user}',
+      config: {
+        auth: { mode: 'try' },
+        handler: (request, reply) => {
+          request.cookieAuth.set({ user: request.params.user });
+          return reply(request.params.user);
+        }
+      }
+    });
+
+    server.route({
+      method: 'GET', path: '/setKey', handler: (request) => {
+        request.cookieAuth.set('key', 'value');
+        done();
+      }
+    });
+
+    server.inject('/login/steve', (res) => {
+      const pattern = /(?:[^\x00-\x20\(\)<>@\,;\:\\"\/\[\]\?\=\{\}\x7F]+)\s*=\s*(?:([^\x00-\x20\"\,\;\\\x7F]*))/;
+      code.expect(res.result).to.equal('steve');
+      const header = res.headers['set-cookie'];
+      code.expect(header.length).to.equal(1);
+      code.expect(header[0]).to.contain('Max-Age=60');
+      const cookie = header[0].match(pattern);
+
+      /* eslint-disable hapi/no-shadow-relaxed */
+      server.inject({ method: 'GET', url: '/setKey', headers: { cookie: `${mainCookie}=${cookie[1]}` } }, (res2) => {
+        code.expect(res2.statusCode).to.equal(200);
+      });
+    });
+  });
+});
+
+lab.test('can use validateFunc to validate', (done) => {
+  config.strategies.session.options.validateFunc = (request, session, callback) => {
```

Comment:
the test should be that it uses a string and calls the server method to validate